### PR TITLE
Add failing test fixture for RemoveParentCallWithoutParentRector

### PR DESCRIPTION
**Pull request by zingimmick**
Just add `@mixin {ParentClass}` to unrelated class and it will throw
```
LogicException : leaveNode() returned invalid value of type integer
```

It works well after removing the `@mixin {ParentClass}`.

### CODE REVIEW

**Comment by samsonasik** [comment]
test passed, did you tried latest rector 0.11.48 ? It possibly already handled at 

https://github.com/rectorphp/rector-src/pull/688
https://github.com/rectorphp/rector-src/pull/689
https://github.com/rectorphp/rector-src/pull/700
https://github.com/rectorphp/rector-src/pull/705

**Comment by zingimmick** [comment]
> test passed, did you tried latest rector 0.11.48 ? It possibly already handled at
> 
> #688
> #689
> #700
> #705

test skipped, let me think about how to reproduce it.

**Comment by zingimmick** [comment]
> test passed, did you tried latest rector 0.11.48 ? It possibly already handled at
> 
> #688
> #689
> #700
> #705

@samsonasik Thanks for reminding, but I think the test passed because of it skipped.

How do I add failing test fixture when the rector test requires PHP < 8.0?
https://github.com/rectorphp/rector-src/blob/c3c1411d107534e9d5c103982c9e5442213cc800/rules-tests/DeadCode/Rector/StaticCall/RemoveParentCallWithoutParentRector/RemoveParentCallWithoutParentRectorTest.php#L13-L17

Temporarily reproduced in the sample: https://github.com/zingimmick/rector-sample/runs/3359280636#step:5:10.

**Comment by TomasVotruba** [comment]
@zingimmick You can configure PHP version in the config for test

**Comment by samsonasik** [comment]
thank you, the parent call without parent class is only working on php < 7.3 iirc, could you provide fix for it? You can check my PRs above for references

**Comment by zingimmick** [comment]
> @zingimmick You can configure PHP version in the config for test

thank you, reproduced now.

**Comment by zingimmick** [comment]
> thank you, the parent call without parent class is only working on php < 7.3 iirc, could you provide fix for it? You can check my PRs above for references

I will try it later.

File: packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php [hunk 1]
```diff
@@ -149,13 +144,17 @@ private function processNodesWithMixinHandling(
...
-        return (bool) $this->betterNodeFinder->findFirst($nodes, function (Node $node) use ($realPath): bool {
-            if (! $node instanceof FullyQualified) {
+        return (bool) $this->betterNodeFinder->findFirst($nodes, function (Node $node): bool {
+
+            if (! $node instanceof FullyQualified && ! $node instanceof Class_) {
                 return false;
             }
 
-            $className = $node->toString();
+            if ($node instanceof Class_ && $node->isAnonymous()) {
```
Comment (samsonasik) [inline packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:152]: you should can use ClassAnalyzer service for detecting anonymous class

File: packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php [hunk 2]
```diff
@@ -147,44 +144,58 @@ private function processNodesWithMixinHandling(
...
+        foreach ($classReflection->getMixinTags() as $mixinTag) {
+            $type = $mixinTag->getType();
+            if (!$type instanceof ObjectType){
                 return false;
             }
 
-            $fileName = $classReflection->getFileName();
-            if ($fileName === false) {
-                return false;
+            if ($type->getClassName() === $classReflection->getName()) {
```
Comment (samsonasik) [inline packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php]: please assign classReflection->getName() to variable before loop to improve performance

File: packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php (hunk 1 above)
Comment (zingimmick) [reply packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:152]: I tried to use ClassAnalyzer service for detecting anonymous class, but it cannot ensure that the property `namespacedName` exists.

Comment (samsonasik) [reply packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:152]: it possibly needs a node instanceof Class_ as well, like this:

```
$node instanceof Class_ && $this->classAnalyzer->isAnonymous()
```

if that still not works, possibly revert is the way for it as it possibly overlapped with filling namespacedName property

Comment (zingimmick) [reply packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:152]: `$this->classAnalyzer->isAnonymous()` will return false like this:
```php
new class extends ParentClass implements SomeInterface
{
    use AnotherTrait;
};
```
but the property namespacedName not exists in this case.

Temporarily reverted to make the test pass.

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @zingimmick

**Pull request merged**